src/nvim/diff.c

```diff
@@ -2626,6 +2925,18 @@ void ex_diffgetput(exarg_T *eap)
   diff_T *dprev = NULL;
 
   for (dp = curtab->tp_first_diff; dp != NULL;) {
+
+    if (dp->is_linematched && !eap->addr_count) {
+      // handle the case with overlapping diff blocks
```

Comment:
yeah, adjacent would be the better way to refer to it. I'm refering to an overlap as the same as adjacent in this case. But technically it is an overlap if you call the end of the diff **line number + count.**
a diff on line 2 of count 2 ends on line (2 + 2 = 4), and if another diff starts on line 4, then they are overlapping. because they are both present on line 4. And that's where all the difficulty has been in getting this to work, handling these cases properly.

Reply:
I'll change it to say adjacent

Reply:
changed